for_workspace/make_build.bzl

```diff
@@ -0,0 +1,47 @@
+""" Rule for building GNU Make from sources. """
+
+load("//tools/build_defs:detect_root.bzl", "detect_root")
+load("@rules_foreign_cc//tools/build_defs:shell_script_helper.bzl", "convert_shell_script")
+
+def _make_tool(ctx):
+    root = detect_root(ctx.attr.make_srcs)
+
+    make = ctx.actions.declare_directory("make")
+    script = [
+        "export BUILD_DIR=##pwd##",
+        "export BUILD_TMPDIR=##tmpdir##",
+        "##copy_dir_contents_to_dir## ./{} $BUILD_TMPDIR".format(root),
+        "cd $$BUILD_TMPDIR$$",
+        "./configure --prefix=$$BUILD_DIR$$/{}".format(make.path),
+        "./build.sh",
+        "./make install",
+    ]
+    script_text = convert_shell_script(ctx, script)
+
+    ctx.actions.run_shell(
+        mnemonic = "BootstrapMake",
+        inputs = ctx.attr.make_srcs.files,
+        outputs = [make],
+        tools = [],
+        use_default_shell_env = True,
+        command = script_text,
+        execution_requirements = {"block-network": ""},
+    )
+
+    return [DefaultInfo(files = depset([make]))]
+
+""" Rule for building Make. Invokes configure script and make install.
+  Attributes:
+    make_srcs - target with the Make sources
+"""
+make_tool = rule(
```

Comment:
Can you add an example to verify manually that it works just like we do for [cmake](https://github.com/bazelbuild/rules_foreign_cc/blob/006d5f6f8b6f6922943add61fcf038a4a1d802e1/examples/build_cmake_itself/BUILD#L3)?

Reply:
Done in 16778ca.

Turns out one needs to install `automake` and `texinfo` on Debian to get Make working, but that should be fine: the intention of this toolchain is to make it replaceable, so that I can swap in my Emscripten wrapper. When the dependencies are installed, the toolchain can be bootstrapped by Bazel.

(Tested on Debian Buster on o ChromeOS device using Crostini.)